src/PoshKentico.Business/Configuration/Users/GetCmsUserBusiness.cs

```diff
@@ -0,0 +1,91 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see &lt;http://www.gnu.org/licenses/&gt;.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using PoshKentico.Core.Services.Configuration.Users;
+
+namespace PoshKentico.Business.Configuration.Users
+{
+    /// <summary>
+    /// Business Layer of the Get-CMSUser cmdlet.
+    /// </summary>
+    [Export(typeof(GetCmsUserBusiness))]
+    public class GetCmsUserBusiness : CmdletBusinessBase
+    {
+        #region Properties
+
+        /// <summary>
+        /// Gets or sets a reference to the User Service. Populated by MEF.
+        /// </summary>
+        [Import]
+        public IUserService UserService { get; set; }
+
+        #endregion
+
+        #region Methods
+
+        /// <summary>
+        /// Gets a list of all of the <see cref="IUser"/> in the CMS System.
+        /// </summary>
+        /// <returns>A list of all of the <see cref="IUser"/>.</returns>
+        public IEnumerable<IUser> GetUsers()
+        {
+            return this.UserService.Users;
+        }
+
+        /// <summary>
+        /// Gets a list of the <see cref="IUser"/> which match the supplied IDs.
+        /// </summary>
+        /// <param name="ids">The IDs of the <see cref="IUser"/> to return.</param>
+        /// <returns>A list of the <see cref="IUser"/> which match the supplied IDs.</returns>
+        public IEnumerable<IUser> GetUsers(params int[] ids)
+        {
+            var users = from id in ids select this.UserService.GetUser(id);
+
+            return (from user in users
+                    where user != null
+                    select user).ToArray();
+        }
+
+        /// <summary>
+        /// Gets a list of all of the <see cref="IUser"/> which match the specified criteria.
+        /// </summary>
+        /// <param name="userName">The user name which to match the users to.</param>
+        /// <param name="exact">A boolean which indicates if the match should be exact.</param>
+        /// <returns>A list of all of the <see cref="IUser"/> which match the specified criteria.</returns>
+        public IEnumerable<IUser> GetUsers(string userName, bool exact)
+        {
+            if (exact)
+            {
+                return (from c in this.UserService.Users
+                        where c.UserName.ToLowerInvariant().Equals(userName, StringComparison.InvariantCultureIgnoreCase)
+                        select c).ToArray();
+            }
+            else
+            {
+                return (from c in this.UserService.Users
+                        where c.UserName.ToLowerInvariant().Contains(userName.ToLowerInvariant())
```

Comment:
Command ```Get-CMSUser *a*``` via a regex isn't working. Might want to perform a regex check here. Similar to:

![image](https://user-images.githubusercontent.com/2670493/48509343-b4c3a000-e805-11e8-8eeb-940e95fa4ee1.png)

Reply:
resolved